Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +26,63 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return totalAmpHoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (current_ < 0)
+       return true;
+    else
+        return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    QTime time(0,0,0,0);
+    int ms_remaining;
+      if (isCharging() == true)
+      {
+        ms_remaining = totalAmpHoursUsed() / averageCurrent_ * MS_CONVERSION;
+      }   else
+      {
+        ms_remaining = (BATTERY_AMP_HOUR_CAPACITY - totalAmpHoursUsed()) / averageCurrent_ * MS_CONVERSION;
+      }
+
+    QTime timeWhenChargedOrDepleted = time.addMSecs(ms_remaining);
+
+    return timeWhenChargedOrDepleted;
+
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // This is where you can update your variables
+    int deltaMseconds;
+    double chargeAh;
+    QTime msTime(0,0,0,0);
```

Comment:
This variable name is also a bit misleading, since it is `batteryData.time`, not just the milliseconds. Perhaps to differentiate it from the previous time, name it something along the lines of `presentTime`